### PR TITLE
fix: Fix action collection not being deleted when the deleted collection in edit mode is published

### LINKED ISSUE #9722: [Bug]: Action collection is not gettting deleted even if we publish deleted actionCollection
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When I'm trying to delete the actionCollection entry I'm unable to do so. Basically `delted` or `deletedAt` fields are not getting updated after we delete the actionCollection from UI. This is causing the issues in import-export flow as this might get exported the deleted actionCollection. 

### Steps To Reproduce

1. Create actionCollection
2. Create binding to a widget on the canvas
3. Publish app
4. Delete the action collection from edit mode
5. `deletedAt` field gets updated in unpublishedCollection as expected
6. Publish the application again
7. Also `deleted` field remains false

### Environment

Release

### Version

Cloud

### PR DESCRIPTION
**Pull request by abhvsn**
## Description

> This will fix the issue related to action collection not being deleted when the deleted collection is published. This fix will fetch the collection by appId and then filter as per the view mode  

Fixes #9722 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Junit TC
> Manual test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ActionCollectionServiceImpl.java [hunk 1]
```diff
@@ -73,7 +73,7 @@ public ActionCollectionServiceImpl(Scheduler scheduler,
 
     @Override
     public Flux<ActionCollection> findAllByApplicationIdAndViewMode(String applicationId, Boolean viewMode, AclPermission permission, Sort sort) {
-        return repository.findByApplicationIdAndViewMode(applicationId, viewMode, permission)
+        return repository.findByApplicationId(applicationId, permission, sort)
```
Comment (mohanarpit) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ActionCollectionServiceImpl.java:76]: Why aren't we considering the `viewMode` any more here? Will this have any other impact?

**Review by nidhi-nair — DISMISSED** [review]

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ActionCollectionServiceImpl.java (hunk 1 above)
Comment (abhvsn) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ActionCollectionServiceImpl.java:76]: We are filtering the collection inside the flatmap for the `viewMode`.

Comment (mohanarpit) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ActionCollectionServiceImpl.java:76]: That doesn't seem to be the case. In the `flatMap` below, we are only checking for `viewMode` value to return Mono.empty() for empty collection sizes. We aren't performing any filtering there.

Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ActionCollectionServiceImpl.java:76]: It's in the upstream method that the filtering is happening @mohanarpit. I think we'll need to go through a cycle of renaming these methods, they are no longer clearly indicative of their purpose.

**Review by nidhi-nair — APPROVED** [review]

**Pull request merged**